### PR TITLE
Small Tweaks: Remove unused variables, unnecessaries parentheses, etc

### PR DESCRIPTION
**Pull request by carusogabriel**
Together with PHPStorm and https://github.com/kalessil/phpinspectionsea, here are some improvements that we can make to Doctrine :smile: 

PS: The commits will be squashed before the merge, they are only this way to inform what was done in each commit.

### CODE REVIEW

**Review by Majkl578 — CHANGES_REQUESTED** [review]
Good overall, just some nitpicks (forward compatibility with new CS) and 3 bugs. :)
Also please squash the changes into one or two commits.

File: lib/Doctrine/ORM/Mapping/AbstractClassMetadataFactory.php [hunk 1]
```diff
@@ -144,7 +144,6 @@ public function getAllMetadata() : array
      * @param string $className The name of the class.
      *
      * @throws \InvalidArgumentException
-     * @throws ReflectionException
```
[the remark this answers is not in the document]
Comment (carusogabriel) [reply lib/Doctrine/ORM/Mapping/AbstractClassMetadataFactory.php]: But the concept of `@throws` annotations arent in the method they happen?

File: lib/Doctrine/ORM/Query/Parser.php [hunk 2]
```diff
@@ -1428,9 +1428,7 @@ public function UpdateItem()
 
         $this->match(Lexer::T_EQUALS);
 
-        $updateItem = new AST\UpdateItem($pathExpr, $this->NewValue());
-
-        return $updateItem;
```
[the remark this answers is not in the document]
Comment (carusogabriel) [reply lib/Doctrine/ORM/Query/Parser.php]: Yes, just need to figure out how. Maybe https://github.com/sirbrillig/phpcs-variable-analysis can help us

File: lib/Doctrine/ORM/QueryBuilder.php [hunk 3]
```diff
@@ -709,7 +709,7 @@ public function add($dqlPartName, $dqlPart, $append = false)
                 $this->dqlParts[$dqlPartName][] = $dqlPart;
             }
         } else {
-            $this->dqlParts[$dqlPartName] = ($isMultiple) ? [$dqlPart] : $dqlPart;
```
[the remark this answers is not in the document]
Comment (carusogabriel) [reply lib/Doctrine/ORM/QueryBuilder.php]: Yeah, we definitely do

File: lib/Doctrine/ORM/AbstractQuery.php [hunk 4]
```diff
@@ -303,7 +303,7 @@ public function getParameter($key)
             function (Query\Parameter $parameter) use ($key) : bool {
                 $parameterName = $parameter->getName();
 
-                return $key === $parameterName || (string) $key === (string) $parameterName;
+                return $key === $parameterName || (string) $key === $parameterName;
```
[the remark this answers is not in the document]
Comment (carusogabriel) [reply lib/Doctrine/ORM/AbstractQuery.php]: Gotcha, thanks!

File: lib/Doctrine/ORM/Query/Parser.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (Majkl578) [reply lib/Doctrine/ORM/Query/Parser.php]: This specific case looks rather easy - if `return` has a single variable, and previous _effective_ statement is an assignment to this variable, then this variable should be useless.

WDYT @kukulich? UselessVariableSniff?

[the remark this answers is not in the document]
Comment (kukulich) [reply lib/Doctrine/ORM/Query/Parser.php]: Interesting idea :) Probably all variables that are used just once could be marked as useless.

**Comment by carusogabriel** [comment]
Anything left to be done here?

**Comment by Ocramius** [comment]
Yes:

 1. rename PR to describe what is being done
 2. smash the commit description into a single line (it fits)

Also: what's missing to get the build green?

**Comment by carusogabriel** [comment]
> What's missing to get the build green?

Strange, it was before I rebase with `master`. Gonna debug what went wrong and fix what you've requested :)

**Comment by carusogabriel** [comment]
Closing as these changes will be add in the next version of https://github.com/doctrine/coding-standard and be done automatically :tada:

**Pull request closed without merging**